xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/DiagramMacros.xml

```diff
@@ -0,0 +1,446 @@
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'xwiki-events-bridge'], function($) {
+  var serviceURL = new XWiki.Document(
+                       XWiki.Model.resolve('Confluence.Macros.DiagramService', XWiki.EntityType.DOCUMENT))
+                       .getURL('get', 'outputSyntax=plain'),
+      createSuccessMessage = "$escapetool.javascript($services.localization.render('confluencediagram.create.success'))",
+      createFailureMessage = "$escapetool.javascript($services.localization.render('confluencediagram.create.error'))",
+      deleteSuccessMessage = "$escapetool.javascript($services.localization.render('confluencediagram.delete.success'))",
+      deleteFailureMessage = "$escapetool.javascript($services.localization.render('confluencediagram.delete.error'))";
+
+  var handleCreateDiagram = function () {
+    $('.create-diagram-btn').on('click', function(e){
+      e.preventDefault();
+
+      var pageReferenceSerialized = $(this).prevAll('.create-diagram-page:first').val(),
+          diagramName = $(this).prevAll('.create-diagram-name:first').val();
+      $.ajax({
+        url: serviceURL,
+        data: {
+          'action' : 'create',
+          'page' : pageReferenceSerialized,
+          'diagramName' : diagramName
+        },
+        success: function(data, status) {
+          new XWiki.widgets.Notification(createSuccessMessage, 'done');
+          location.reload();
+        },
+        error: function() {
+          new XWiki.widgets.Notification(createFailureMessage, 'error');
+        }
+      });
+    });
+  }
+
+  var handleDeleteDiagram = function() {
+    $('.delete-diagram-btn').on('click', function(e){
+      e.preventDefault();
+
+      var pageReferenceSerialized = $(this).prevAll('.delete-diagram-page:first').val();
+      $.ajax({
+        url: serviceURL,
+        data: {
+          'action' : 'delete',
+          'page' : pageReferenceSerialized
+        },
+        success: function(data, status) {
+          new XWiki.widgets.Notification(deleteSuccessMessage, 'done');
+          location.reload();
+        },
+        error: function() {
+          new XWiki.widgets.Notification(deleteFailureMessage, 'error');
+        }
+      });
+    });
+  }
+
+  var init = function() {
+    handleCreateDiagram();
+    handleDeleteDiagram();
+  }
+  XWiki.domIsLoaded &amp;&amp; init();
```

Comment:
```suggestion
  $(init);
```

Don't you need to also re-initialize on ``xwiki:dom:updated``?

    $(document).on('xwiki:dom:updated', init);

Reply:
Fixed